test/e2e/bundle_factory.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+type bundleFactory struct {
+	assets       *assets
+	t            *testing.T
+	bundleFolder string
+}
+
+func newBundleDir(t *testing.T, assets *assets) bundleFactory {
+	return bundleFactory{assets: assets, t: t}
+}
+
+func (b *bundleFactory) isImageABundle(imgRef string) bool {
```

Comment:
this method seems to be extracted only for two tests: TestBundlePushPullAnnotation and TestPushWithFileExclusion. arguably TestPushWithFileExclusion shouldnt be testing that image is a bundle (covered already in first test), hence i am thinking it might be better to inline this into TestBundlePushPullAnnotation and remove reference to that in TestPushWithFileExclusion.